app/javascript/components/patient/close_contacts/CloseContactTable.js

```diff
@@ -0,0 +1,489 @@
+      }
+    );
+  };
+
+  /**
+   * Called when the user clicks on the Contact Attempt for a Close Contact
+   */
+  handleContactAttempt = async data => {
+    if (await confirmDialog('Are you sure you want to log an additional contact attempt?', { title: 'New Contact Attempt' })) {
+      let currentCC = _.cloneDeep(this.state.table.rowData[`${data}`]);
+      currentCC['contact_attempts'] = currentCC['contact_attempts'] + 1;
+      this.setState(
+        {
+          showEditModal: false,
+          editRow: null,
+        },
+        () => {
+          this.updateCloseContact(currentCC, true);
+        }
+      );
+    }
+  };
+
+  /**
+   * Called when a page is clicked in the pagination component.
+   * Updates the table based on the selected page.
+   * @param {Object} page - Page object from react-paginate
+   */
+  handlePageUpdate = page => {
+    this.setState(
+      state => {
+        return {
+          query: { ...state.query, page: page.selected },
+        };
+      },
+      () => {
+        this.updateTable(this.state.query);
+      }
+    );
+  };
+
+  /**
+   * Toggle the Add New Close Contact modal by updating state.
+   */
+  toggleAddModal = () => {
+    this.setState({
+      showAddCloseContactModal: !this.state.showAddCloseContactModal,
+    });
+  };
+
+  /**
+   * Closes the Add New Close Contact modal and makes a request to add a new close contact to the db.
+   * @param {*} newCloseContactData - State from close contact modal containing needed close contact data.
+   */
+  handleAddSubmit = newCloseContactData => {
+    this.setState({ showAddCloseContactModal: false }, () => {
+      this.updateCloseContact(newCloseContactData, false);
+    });
+  };
+
+  /**
+   * Called when the edit close contact button is clicked.
+   * Updates the state to show the appropriate modal for editing a close contact.
+   */
+  toggleEditModal = row => {
+    this.setState({
+      showEditModal: true,
+      editRow: row,
+    });
+  };
+
+  /**
+   * Closes the Edit Close Contact modal and makes a request to update an existing close contact record.
+   * @param {*} updatedCloseContactData - State from close contact modal containing updated close contact data.
+   */
+  handleEditSubmit = updatedCloseContactData => {
+    const currentCloseContactId = this.state.table.rowData[this.state.editRow]?.id;
+    updatedCloseContactData['id'] = currentCloseContactId;
+    this.setState(
+      {
+        showEditModal: false,
+        editRow: null,
+      },
+      () => {
+        this.updateCloseContact(updatedCloseContactData, true);
+      }
+    );
+  };
+
+  // Toggles the delete modal and clears any reason or reason text
+  toggleDeleteModal = id => {
+    let selectedCloseContactID = _.isNumber(id) ? id : null;
+    this.setState({
+      showDeleteModal: !this.state.showDeleteModal,
+      selectedCloseContactID,
+      delete_reason: null,
+      delete_reason_text: null,
+    });
+  };
+
+  handleDeleteSubmit = () => {
```

Comment:
seeing this error on delete

<img width="553" alt="Screen Shot 2021-07-09 at 9 11 06 AM" src="https://user-images.githubusercontent.com/35042815/125082803-a4872000-e095-11eb-9734-4953fc9af3e0.png">

Reply:
This was due to me incorrectly changing `onChange(myFunc)` to `onChange(() => myFunc)`